clients/js-legacy/src/extensions/confidentialTransfer/elgamal.ts

```diff
@@ -0,0 +1,61 @@
+import { blob } from '@solana/buffer-layout';
+import type { Layout } from '@solana/buffer-layout';
+import { encodeDecode } from '@solana/buffer-layout-utils';
+import { PodElGamalPubkey, PodElGamalCiphertext, PodAeCiphertext } from '@solana/zk-sdk';
+
+export const elgamalPublicKey = (property?: string): Layout<PodElGamalPubkey> => {
+    const layout = blob(32, property);
+    const { encode, decode } = encodeDecode(layout);
+
+    const elgamalPublicKeyLayout = layout as Layout<unknown> as Layout<PodElGamalPubkey>;
+
+    elgamalPublicKeyLayout.decode = (buffer: Buffer, offset: number) => {
+        const src = decode(buffer, offset);
+        return new PodElGamalPubkey(src);
+    };
+
+    elgamalPublicKeyLayout.encode = (elgamalPublicKey: PodElGamalPubkey, buffer: Buffer, offset: number) => {
+        const src = elgamalPublicKey.toBytes();
+        return encode(src, buffer, offset);
+    };
+
+    return elgamalPublicKeyLayout;
+};
+
+export const elgamalCiphertext = (property?: string): Layout<PodElGamalCiphertext> => {
+    const layout = blob(32, property);
```

Comment:
Shouldn't this be 64? In the ZK SDK, `ELGAMAL_CIPHERTEXT_LEN` has the decrypt handle and the pedersen commitment

Reply:
Ah nice catch! Fixed!